R/dsl.R

```diff
@@ -146,6 +148,63 @@ do_package_checks <- function(...,
       add_code_step(covr::codecov(quiet = FALSE))
   }
 }
+#' @description
+#' `do_pkgdown_site()` builds and optionally deploys a pkgdown site and adds default steps
+#'   to the `"install"`, `"before_deploy"` and `"deploy"` stages:
+#'
+#' @inheritParams step_build_pkgdown
+#' @inheritParams step_setup_push_deploy
+#' @inheritParams step_do_push_deploy
+#' @param build_only Build the pkgdown site but do not deploy it.
+#' @param deploy Checks if env variable `id_rsa` is set in Travis using [ci_can_push()]. If missing,
+#'   deployment is not possible.
+#'
+#' @rdname DSL
+#' @export
+#' @importFrom magrittr %>%
+do_pkgdown <- function(...,
+                       build_only = FALSE,
+                       deploy = ci_can_push(),
```

Comment:
Why do we need two arguments here? Wouldn't only a `deploy` argument suffice?

Reply:
Hm yes, looks like they are redundant! The check whether `id_rsa` exists can be done internally.